apple/internal/apple_framework_import.bzl

```diff
@@ -314,12 +344,95 @@ def _debug_info_binaries(
 
     return all_binaries_dict.values()
 
-def _apple_dynamic_framework_import_impl(ctx):
-    """Implementation for the apple_dynamic_framework_import rule."""
+def _get_current_library_identifier(
+        *,
+        current_platform,
+        xcframework_path,
+        xcframework_imports):
+    library_identifiers = sets.make()
+
+    xcframework_path_length = len(xcframework_path)
+    for f in xcframework_imports:
+        inner_path = f.path[xcframework_path_length + 1:]
+        for i in range(len(inner_path)):
+            if inner_path[i] == "/":
+                identifier = inner_path[:i]
+                sets.insert(library_identifiers, identifier)
+                break
+
+    platform_type = str(current_platform.platform_type).lower()
+    is_device = current_platform.is_device
+    for id in sets.to_list(library_identifiers):
+        if not id.startswith(platform_type):
+            continue
+        if is_device and (id.endswith("-simulator") or id.endswith("-maccatalyst")):
+            continue
+        elif id.endswith("-simulator"):
+            return id
```

Comment:
There was some reluctance in the past about how these identifiers are completely arbitrary, so realistically reading the plist is the only fully correct way to get this info, but I think we can acknowledge that and accept this limitation